packages/user_packages/iotcc_mqtt.py

```diff
@@ -31,9 +31,36 @@
 # ----------------------------------------------------------------------------#
 
 from liota.core.package_manager import LiotaPackage
+import time
+import logging
+import random
+
+log = logging.getLogger(__name__)
 
 dependencies = ["general_edge_system", "credentials"]
 
+# --------------------User Configurable Retry and Delay Settings------------------------------#
+
+# The value mentioned below is in multiple of 1K, 2 for 2K, 3 for 3K, 4 for 4K and  5 for 5K Edge Systems
+no_of_edge_system_in_thousands = 1
+# Number of Retries for Connection and Registrations
+no_of_retries_for_connection = 5
+# MQTT Connection Delay Min Value in seconds
+mqtt_connection_retry_delay_min = 10
+# MQTT Connection Delay Max Value in seconds
+mqtt_connection_retry_delay_max = 600
+# Retry delay Min Value in seconds
+delay_retries_min = 600
+# Retry delay Max Value in seconds
+retries_delay_max = 1800
+
+# Lambda Function Multiplier uses the above settings for calculating retry and delay logic
+lfm = lambda x: x * no_of_edge_system_in_thousands
+retry_attempts = lfm(no_of_retries_for_connection)
+mqtt_connection_delay_retries = random.randint(lfm(mqtt_connection_retry_delay_min),
```

Comment:
mqtt_connection_delay_retries is used only once hence it is okay to be present at the top of the package.

delay_retries are used at multiple places but the calls are independent hence it should be okay to have them common. Also, the randomness is kind of estimation to avoid multiple packages retrying the same time from different edge_system which is addressed with the randomness provided by the above values.